cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/config/KafkaCruiseControlConfig.java

```diff
@@ -1744,11 +1775,28 @@
                 300000L,
                 ConfigDef.Importance.LOW,
                 ANOMALY_DETECTION_INTERVAL_MS_DOC)
+        .define(GOAL_VIOLATION_DETECTION_INTERVAL_MS_CONFIG,
+                ConfigDef.Type.LONG,
+                ConfigDef.Importance.LOW,
+                GOAL_VIOLATION_DETECTION_INTERVAL_MS_DOC)
+        .define(METRIC_ANOMALY_DETECTION_INTERVAL_MS_CONFIG,
```

Comment:
Should we add a sanity check to ensure that the value of this config is less frequent compared to the sampling? -- i.e. otherwise it is no-op.

Reply:
add the sanity check.